### ISSUE TITLE
[Question] callback functionality like SB3?

### ISSUE THREAD

**Issue by emrul** [aai-institute/tianshou#854]
- [X] I have marked all applicable categories:
    + [-] exception-raising bug
    + [-] RL algorithm bug
    + [X] documentation request (i.e. "X is missing from the documentation.")
    + [X] new feature request
- [X] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [X] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [-] I have mentioned version numbers, operating system and environment, where applicable:

I have a question about migrating some RL stuff from SB3 to Tianshou.  In SB3 I am able to define a [callback](https://stable-baselines3.readthedocs.io/en/master/guide/callbacks.html), that, during evaluation in a training loop extracts out additional metrics from the environment's info object for all completed environments and [logs](https://stable-baselines3.readthedocs.io/en/master/guide/tensorboard.html#logging-more-values) them to Tensorboard.

I've been searching for an elegant way to do the same in Tianshou but I can't seem to locate a good/clean way of doing it. Is there any documentation or example on how I can do something similar?

**Comment by Trinkle23897** [comment]
Probably something similar to https://tianshou.readthedocs.io/en/master/tutorials/cheatsheet.html#handle-batched-data-stream-in-collector
https://github.com/thu-ml/tianshou/blob/e7c2c3711eb4f4430ff36e6df1cf8630714815ad/docs/tutorials/cheatsheet.rst?plain=1#L223-L253

and route Logger into your VenvWrapper (Logger is an attribute in VenvWrapper). In general, we prefer subclass over adding hook to adding new features.

**Comment by emrul** [comment]
This could work quite well - thank you @Trinkle23897 - I will try it this week!

**Comment by emrul** [comment]
Just to close the loop on this one - I wanted to keep some additional statistics across batches of episodes. I ended up extending the `Collector` and the `TensorboardLogger` which did the trick for me. While I was at it, I renamed the log variables to match SB3 so I could compare easily with the results from my SB3 implementation.

So far so good, it took me a while to understand the particular parameters (especially around batch, epoch, num_steps, etc.) but once I was patient to read the docs and check how the code is implemented it wasn't too hard.
